src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -185,6 +201,11 @@ protected override async Task RemoveKeyAsync(string cacheKey, CacheSerializerHin
                         Logger.BackPropagateL2toL1(_logger, memoryCacheEntryOptions.Size ?? 0);
                         _memoryCache.Set(cacheKey, result, memoryCacheEntryOptions);
                         Logger.MemoryCacheCount(_logger, _memoryCacheType, read, _memoryCache.Count);
+
+                        if (telemetryData != null)
+                        {
+                            telemetryData.CacheLevel = Client.Cache.CacheLevel.L2Cache;
```

Comment:
>telemetryData [](http://example.com/codeflow?start=28&length=13)

 I would rather not take this now, as we still have not figured out our client side telemetry story and would prefer to not have API changes which we cannot remove later.

Reply:
Are you objecting to how the data is collected or to what data is collected? I agree with the previous comment on the "HOW".

But on the WHY, we have agreed with E&D team to deliver this, in a meeting facilitated by @TimHannMSFT . The spec is [here](https://microsoft-my.sharepoint-df.com/:w:/p/nebharg/EWrRAFHR6VdGvx5i_-uCZiUBROlxzCXZzLdRUJ8ePTKkgw?e=bHx3KO)